src/Core/Settings.h

```diff
@@ -350,6 +350,7 @@ class IColumn;
     M(Bool, opentelemetry_trace_processors, false, "Collect OpenTelemetry spans for processors.", 0) \
     M(Bool, prefer_column_name_to_alias, false, "Prefer using column names instead of aliases if possible.", 0) \
     M(Bool, allow_experimental_analyzer, false, "Allow experimental analyzer", 0) \
+    M(Bool, allow_get_http_header, false, "Allow function getClientHTTPHeader", 0) \
```

Comment:
also needs to be a server setting, we don't want to allow users to get around this by simply modifying the setting.